### ISSUE TITLE
lua-mastodon failure to install on termux

### ISSUE THREAD

**Issue by vielmetti** [hishamhm/lua-mastodon#6]
This is probably a problem upstream, but I thought I'd report it here to start the discussion.

I'm trying to install lua-mastodon on Android under Termux. I get this error message of an undefined dependency.
```
Installing https://luarocks.org/lua-mastodon-0.1-1.src.rock
Missing dependencies for lua-mastodon 0.1-1:
   luafilesystem (not installed)
   lua-requests (not installed)
   split (not installed)
   luasocket (not installed)
   luasec (not installed)
   date (not installed)
   mimetypes (not installed)

lua-mastodon 0.1-1 depends on luafilesystem (not installed)
Installing https://luarocks.org/luafilesystem-1.6.3-2.src.rock
gcc -O2 -fPIC -I/data/data/com.termux/files/usr/include -c src/lfs.c -o src/lfs.o
src/lfs.c:63:10: fatal error: 'lua.h' file not found
#include <lua.h>
         ^~~~~~~
1 error generated.

Error: Failed installing dependency: https://luarocks.org/luafilesystem-1.6.3-2.src.rock - Build error: Failed compiling object src/lfs.o
$```

I'll find the upstream and report there too. If you haven't used Termux, it's a great shell for Android.

**Comment by hishamhm** [comment]
Your LuaRocks installation is failing to find the Lua headers.

1. Make sure you have the Lua headers installed (i.e. the package that provides the file `lua.h`)
2. Make sure LuaRocks knows where the Lua headers are
2.1. if you installed LuaRocks by hand the `./configure` script should check this (and if it didn't you should file an issue with LuaRocks)
2.2. if you used a package manager, it is probably misconfiguring LuaRocks (and you should file an issue with the package provider).

To workaround the issue, type `luarocks` with no arguments and see what is the path of the `User config` file. Then edit/create that file and add `variables = { LUA_INCDIR = "/path/to/the/directory/where/lua.h/resides" }`

This is really a Lua/LuaRocks installation issue not specific to lua-mastodon, so I'm closing this one here.

**Comment by vielmetti** [comment]
Problem improved, but not solved.

First I looked at http://stackoverflow.com/questions/18667112/using-mysql-in-lua-script

then I did

```
packages install liblua-dev
```

which pulled in all the dev files except for not being able to build `luasec` which depends on OpenSSL.

I'll try to find the right upstream to report this / chase this at, hope you don't mind that I document the process here in this closed item for the next person along.
